test_tortoise_interpreter.py

```diff
@@ -58,22 +67,28 @@ def test_sanitized_programs(program, expected):
 
 
 def test_navigate_command_functions():
-    def pen_up(command):
-        return f'PEN UP'
-    
-    def pen_colour(command):
-        return f'P {command[1]}'
-
-    COMMANDS = {'U': pen_up, 'P': pen_colour}
-    program = [['P', '3'], ['U']]
-    command_to_parse = navigate_command_functions(COMMANDS, program)
 
-    assert command_to_parse == ['1. P 3', '2. PEN UP']
+    COMMANDS = {'U': pen_up,'D': pen_down ,'P': pen_colour, 'N': move_north, 'E': move_east}
+    program = [['P', '3'], ['U'], ['N', '1'], ['U'], ['D'], ['E', '2'], ['D']]
+    command_to_parse = navigate_command_functions(COMMANDS, program)
 
+    assert command_to_parse == ['1. P 3',
+                                '2. PEN UP',
+                                '3. Move 1 unit to the north.',
+                                '4. PEN UP (ignored)',
+                                '5. PEN DOWN',
+                                '6. Draw a line 2 units to the east.',
+                                '7. PEN DOWN (ignored)',
+                                ]
 
 def test_pen_up():
-    assert pen_up(['U']) == 'PEN UP'
+    assert pen_up(['U'], 'UP', 'DOWN') == 'PEN UP'
```

Comment:
I know that `test_pen_up` was already here. However, I don't think it's a good idea to test low level details. You don't have to take them out, but maybe don't add any more.

Reply:
removed